src/main/java/Duke.java

```diff
@@ -1,10 +1,283 @@
+                        }
+                        String description = parts[0].trim();
+                        String[] eventParts = parts[1].split("/to");
+                        if (eventParts.length != 2 || eventParts[0].trim().isEmpty() || eventParts[1].trim().isEmpty()) {
+                            throw new DukeException("Invalid command format. Please use 'event <description> /from <start time> /to <end time>'.");
+                        }
+                        String from = eventParts[0].trim();
+                        String to = eventParts[1].trim();
+                        LocalTime startTime = LocalTime.parse(from, DateTimeFormatter.ofPattern("HHmm"));
+                        LocalTime endTime = LocalTime.parse(to, DateTimeFormatter.ofPattern("HHmm"));
+                        tasks.add(new EventTask(description, startTime, endTime));
+                        System.out.println("Got it. I've added this task:");
+                        System.out.println(tasks.get(tasks.size() - 1));
+                        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                    } else {
+                        throw new DukeException("I'm sorry, but I don't know what that means :-(");
+                    }
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Invalid command format. Please provide a valid task index.");
+            } catch (DukeException e) {
+                System.out.println("____________________________________________________________");
+                System.out.println("OOPS!!! " + e.getMessage());
+                System.out.println("____________________________________________________________");
+            }
+            System.out.println("____________________________________________________________");
+            command = scanner.nextLine();
+        }
+        duke.saveTasksToFile(tasks);
+    
+        System.out.println("____________________________________________________________");
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("____________________________________________________________");
+    }
+    
+    
+    private ArrayList<Task> loadTasksFromFile() {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            File file = new File(FILE_PATH);
+            if (!file.exists()) {
+                file.createNewFile();
+            } else {
+                Scanner scanner = new Scanner(file);
+                while (scanner.hasNextLine()) {
+                    String line = scanner.nextLine();
+                    Task task = parseTaskFromString(line);
+                    tasks.add(task);
+                }
+                scanner.close();
+            }
+        } catch (IOException e) {
+            System.out.println("Error loading tasks from file: " + e.getMessage());
```

Comment:
Can consider making this into your own exception class and include the additional error messages as part of the new exception